### PR TITLE
[INLONG-8959][Sort] Add Mongodb-cdc-connector on flink1.15

### LINKED ISSUE #8959: [Feature][Sort]Add mongodb connector on flink 1.15
### Description

Parent issue:[#8336 ](8336)
Add mongodb connector on flink 1.15

### Use case

Add mongodb connector on flink 1.15

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Curry-CPU**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #8959 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/pom.xml [hunk 1]
```diff
@@ -47,6 +47,17 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>mongodb</artifactId>
+            <version>${testcontainers.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.mongodb</groupId>
+            <artifactId>mongodb-driver-sync</artifactId>
+            <version>4.10.2</version>
```
Comment (vernedeng) [inline inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/pom.xml:58]: make this version

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mongodb-cdc/pom.xml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+                                </filter>
+                            </filters>
+                            <relocations>
+                                <relocation>
+                                    <pattern>org.apache.inlong.sort.base</pattern>
+                                    <shadedPattern>org.apache.inlong.sort.cdc.mongodb.shaded.org.apache.inlong.sort.base</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>org.apache.kafka</pattern>
+                                    <shadedPattern>com.ververica.cdc.connectors.shaded.org.apache.kafka</shadedPattern>
```
Comment (vernedeng) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mongodb-cdc/pom.xml:134]: org.apache.inlong.sort.XXXX

File: inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/pom.xml (hunk 1 above)
Comment (Curry-CPU) [reply inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/pom.xml:58]: ok,I will fix it

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mongodb-cdc/pom.xml [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+                                </filter>
+                            </filters>
+                            <relocations>
+                                <relocation>
+                                    <pattern>org.apache.inlong.sort.base</pattern>
+                                    <shadedPattern>org.apache.inlong.sort.cdc.mongodb.shaded.org.apache.inlong.sort.base</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>org.apache.kafka</pattern>
+                                    <shadedPattern>corg.apache.inlong.sort.shaded.org.apache.kafka</shadedPattern>
```
Comment (vernedeng) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mongodb-cdc/pom.xml:134]: Relocation needs the same root dictionary like

org.apache.inlong.sort.cdc.mongodb.shaded

File: inlong-distribution/src/main/assemblies/sort-connectors-v1.15.xml [hunk 4]
```diff
@@ -67,6 +75,14 @@
             </includes>
             <fileMode>0644</fileMode>
         </fileSet>
+        <fileSet>
+            <directory>../inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mongodb-cdc/target</directory>
```
Comment (EMsnap) [inline inlong-distribution/src/main/assemblies/sort-connectors-v1.15.xml:79]: need two dependencies ?

File: inlong-sort/sort-core/pom.xml [hunk 5]
```diff
@@ -269,6 +275,12 @@
                     <version>${project.version}</version>
                     <scope>test</scope>
                 </dependency>
+                <dependency>
```
Comment (EMsnap) [inline inlong-sort/sort-core/pom.xml:278]: ditto

File: inlong-distribution/src/main/assemblies/sort-connectors-v1.15.xml (hunk 4 above)
Comment (Curry-CPU) [reply inlong-distribution/src/main/assemblies/sort-connectors-v1.15.xml]: sorry， I have deleted it

File: inlong-sort/sort-core/pom.xml (hunk 5 above)
Comment (Curry-CPU) [reply inlong-sort/sort-core/pom.xml]: sorry， I have deleted it

**Review by EMsnap — APPROVED** [review]

**Review by e-mhui — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**